libs/cart/src/drivers/magento/transforms/outputs/cart-coupon-response.service.ts

```diff
@@ -0,0 +1,84 @@
+import { Injectable } from '@angular/core';
+
+import { DaffMagentoCartItemTransformer } from './cart-item.service';
+import { MagentoCart } from '../../models/outputs/cart';
+import { DaffCart } from '../../../../models/cart';
+import { daffMagentoCouponTransform } from './cart-coupon';
+
+/**
+ * Transforms magento carts into an object usable by daffodil.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffMagentoCartCouponResponseTransformer {
```

Comment:
I think we've been trying to move toward these being pure functions rather than injectable services, since they no longer need/can be paired with an injection token. Could you change this?

Reply:
@lderrickable I need to inject the cart item transformer. Is there a way to do this with a pure function?

Reply:
oh right. No, I think needs to be done after the item transformer is a pure function too.